src/Switch.tsx

```diff
@@ -0,0 +1,358 @@
+        outline-offset: 2px;
+
+        .Toggle-knob {
+          left: -1px;
+          top: -1px;
+          bottom: -1px;
+          transform: translateX(${props.on ? 'calc(100% + 1px)' : '0'});
+        }
+      `
+    }
+  }}
+
+  ${props => {
+    if (!props.disabled) {
+      if (props.on) {
+        return css`
+          &:hover,
+          &:focus:focus-visible {
+            :after {
+              background-color: ${get('colors.accent.muted')};
+              opacity: 0.2;
+            }
+          }
+
+          &:active,
+          &:active:focus-visible {
+            :after {
+              background-color: ${get('colors.accent.muted')};
+              opacity: 0.5;
+            }
+          }
+        `
+      } else {
+        return css`
+          &:hover,
+          &:focus:focus-visible {
+            .Toggle-knob {
+              background-color: ${get('colors.btn.hoverBg')};
+              border-color: ${get('colors.border.default')};
+            }
+          }
+
+          &:active,
+          &:active:focus-visible {
+            .Toggle-knob {
+              background-color: ${get('colors.btn.activeBg')};
+            }
+          }
+        `
+      }
+    }
+  }}
+
+  &:disabled {
+    cursor: not-allowed;
+  }
+
+  ${sx}
+  ${sizeVariants}
+`
+
+const hiddenTextStyles: BetterSystemStyleObject = {
+  visibility: 'hidden',
+  height: 0
+}
+
+const Switch: React.FC<SwitchProps> = ({
+  'aria-labelledby': ariaLabelledby,
+  'aria-describedby': ariaDescribedby,
+  defaultOn,
+  disabled,
+  isLoading,
+  on: onProp,
+  onChange,
+  onClick,
+  size,
+  statusLabelPosition,
+  sx: sxProp
+}) => {
+  const isControlled = typeof defaultOn === 'undefined' && typeof onProp !== 'undefined'
+  const {theme, colorScheme} = useTheme()
+  const [onState, setOnState] = useProvidedStateOrCreate<boolean>(onProp, onChange, Boolean(defaultOn))
+  const acceptsInteraction = !disabled && !isLoading
+  const handleToggleClick: MouseEventHandler = useCallback(
+    e => {
+      if (!isControlled) {
+        setOnState(!onState)
+      }
+      onClick && onClick(e)
+    },
+    [onClick, isControlled, onState, setOnState]
+  )
+
+  useEffect(() => {
+    if (onChange && isControlled) {
+      onChange(Boolean(onProp))
+    }
+  }, [onChange, onProp, isControlled])
+
+  return (
+    <Box
+      display="inline-flex"
+      alignItems="center"
+      flexDirection={statusLabelPosition === 'start' ? 'row' : 'row-reverse'}
+      mb={4}
+      sx={sxProp}
+    >
+      {isLoading ? <Spinner size="small" /> : null}
+      <Text
+        color={acceptsInteraction ? 'fg.default' : 'fg.muted'}
+        fontSize={size === 'small' ? 0 : 1}
+        mx={2}
+        aria-hidden="true"
+        sx={{position: 'relative'}}
+      >
+        <Box textAlign="right" sx={onState ? null : hiddenTextStyles}>
+          On
+        </Box>
+        <Box textAlign="right" sx={onState ? hiddenTextStyles : null}>
+          Off
+        </Box>
```

Comment:
Is there a reason something like this wouldn't work?
```suggestion
        <Box textAlign="right">
          {isOn ? 'On' : 'Off'}
        </Box>
```

Reply:
I always want the label to take up the same amount of space whether it's "on" or "off". This isn't much an an issue for English, but it could be for other languages.

For example, French:
on = "au"
off = "désactivé"